codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/testutil/Rust.kt

```diff
@@ -55,8 +56,14 @@ private fun tempDir(directory: File? = null): File {
  * This workspace significantly improves test performance by sharing dependencies between different tests.
  */
 object TestWorkspace {
-    private val baseDir =
-        System.getenv("SMITHY_TEST_WORKSPACE")?.let { File(it) } ?: tempDir()
+    private val baseDir by lazy {
+        val homeDir = System.getProperty("user.home")
+        if (homeDir != null) {
+            File(Path.of(homeDir, ".smithy-test-workspace").absolutePathString())
```

Comment:
Can this be `$XDG_DATA_HOME`? Unsure if something like [`dirs::data_local_dir`](https://docs.rs/dirs/latest/dirs/fn.data_local_dir.html) exists for Kotlin. A directory under `/tmp/` might also be a good candidate for where to place this workspace. The home directory is too invasive.

Reply:
Note that if we use `XDG_DATA_HOME`, we should default to [`$HOME/.local/share` if it's not set](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).
Also, if we use this, should we remove the usage of `SMITHY_TEST_WORKSPACE`?

Reply:
good idea—Updated to match dirs::data_local_dir. I kept the fallback, might be useful for CI